supabase/models/filtering-items/searchItems.ts

```diff
@@ -0,0 +1,62 @@
+import { PartialItem } from '@/types/supabaseTypes';
+import newClient from '../../utils/newClient';
+import { SearchParamsType } from '@/types/searchPageTypes';
+
+export default async function searchItems(
+  searchParams: SearchParamsType
+): Promise<PartialItem[]> {
+  const { query, category, subcategory, limit, cursor } = searchParams;
+  const supabase = newClient();
+
+  try {
+    let queryBuilder = supabase.from('items').select();
+
+    switch (true) {
+      case query && !!category && !!subcategory:
```

Comment:
I think we went over this syntax before but I'm struggling a little.

I suppose it's checking if cases evaluate to `true`, which feels weird but I can see how it fits our needs here. But why are we double-negating if just say `category &&` will evaluate it to a truthy if there is a value to the variable?